app/javascript/components/subject/CaseStatus.js

```diff
@@ -73,6 +73,12 @@ class CaseStatus extends React.Component {
 
   submit() {
     let diffState = Object.keys(this.state).filter(k => _.get(this.state, k) !== _.get(this.origState, k));
+
+    // force last public health action to update
+    if (this.state.case_status === 'Suspect' || this.state.case_status === 'Unknown' || this.state.case_status == 'Not a Case') {
```

Comment:
This check is done 4 times in this component. Rather than having to update in all of those places if this ever changes, I think we can consolidate. How about we add a state variable that tracks this condition - name it something appropriate like `shouldBeMoved` (or whatever makes the most sense) and set that based on the starting case status. When the case status changes, in `handleChange()` we can update `this.state.public_health_action` based on this variable and that way we don't have to check again here as it will already be in `diffState`

Reply:
@holmesie what would you call this variable?

Reply:
like workflow change? or something like that

Reply:
@kylietmo refactored but found I didn't really need to store that variable in the state. lmk what you think

Reply:
This helps remove one duplicated check, I had suggested storing a variable in state because the same check is also done on lines 110 and 195

Reply:
done